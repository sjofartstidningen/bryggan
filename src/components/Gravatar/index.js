import React from 'react';
import PropTypes from 'prop-types';
import md5 from 'md5';

function Gravatar({ email, alt, className, onLoad, onError }) {
  const hash = md5(email);
  const src = `https://www.gravatar.com/avatar/${hash}`;

  return (
    <img
      src={src}
      alt={alt}
      className={className}
      onLoad={onLoad}
      onError={onError}
    />
  );
}

Gravatar.propTypes = {
  email: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  className: PropTypes.string,
  onLoad: PropTypes.func,
  onError: PropTypes.func,
};

Gravatar.defaultProps = {
  className: undefined,
  onLoad: undefined,
  onError: undefined,
};

export default Gravatar;
