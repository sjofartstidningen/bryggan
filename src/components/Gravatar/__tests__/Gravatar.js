import React from 'react';
import { shallow } from 'enzyme';
import md5 from 'md5';
import Gravatar from '../';

describe('Component.Gravatar', () => {
  test('Should hash email and request image from Gravatar', () => {
    const email = 'adam@fransvilhelm.com';
    const wrapper = shallow(<Gravatar email={email} alt="" />);
    expect(wrapper).toMatchSnapshot();

    let img = wrapper.find('img');
    let expectedSrc = `https://www.gravatar.com/avatar/${md5(email)}`;

    expect(img.props().src).toEqual(expectedSrc);

    const newEmail = 'random@email.com';
    wrapper.setProps({ email: newEmail });
    wrapper.update();

    img = wrapper.find('img');
    expectedSrc = `https://www.gravatar.com/avatar/${md5(newEmail)}`;

    expect(img.props().src).toEqual(expectedSrc);
  });

  test('should accept alt and className', () => {
    const wrapper = shallow(
      <Gravatar email="random@email.com" alt="Profile picture" className="profile-image" />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should call onLoad-prop when image is loaded', () => {
    const email = 'adam@fransvilhelm.com';
    const handleLoad = jest.fn();
    const wrapper = shallow(
      <Gravatar
        email={email}
        alt=""
        className="profile-image"
        onLoad={handleLoad}
      />,
    );

    const img = wrapper.find('.profile-image');
    img.simulate('load', { loaded: true });

    expect(handleLoad).toHaveBeenCalled();
  });

  test('should call onError-prop when loading image fails', () => {
    const email = 'adam@fransvilhelm.com';
    const handleError = jest.fn();
    const wrapper = shallow(
      <Gravatar
        email={email}
        alt=""
        className="profile-image"
        onError={handleError}
      />,
    );

    const img = wrapper.find('.profile-image');
    img.simulate('error', { error: true });

    expect(handleError).toHaveBeenCalled();
  });
});
