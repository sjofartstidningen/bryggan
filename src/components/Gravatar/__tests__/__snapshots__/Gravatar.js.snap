// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component.Gravatar Should hash email and request image from Gravatar 1`] = `
<img
  alt=""
  src="https://www.gravatar.com/avatar/a3aacee57e15dff19b19ae7bec1ff4b2"
/>
`;

exports[`Component.Gravatar should accept alt and className 1`] = `
<img
  alt="Profile picture"
  className="profile-image"
  src="https://www.gravatar.com/avatar/676d90a1574e9d3ebf98dd36f7adad60"
/>
`;
